source/server/admin/clusters_handler.cc

```diff
@@ -100,7 +100,8 @@ void setHealthFlag(Upstream::Host::HealthFlag flag, const Upstream::Host& host,
 // TODO(efimki): Add support of text readouts stats.
 void ClustersHandler::writeClustersAsJson(Buffer::Instance& response) {
   envoy::admin::v3::Clusters clusters;
-  for (const auto& [name, cluster_ref] : server_.clusterManager().clusters()) {
+  auto all_clusters = server_.clusterManager().clusters();
```

Comment:
Can you move the TODO about outputing warming clusters from the CM code to here?

Reply:
done